### PR TITLE
configuration: add document for the log.timeout configuration

### PR DESCRIPTION
**Pull request by tiancaiamao**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)

TiDB v7.1.0 introduces a new configuration item `log.timeout`.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v7.2 (TiDB 7.2 versions)
- [x] v7.1 (TiDB 7.1 versions)
- [ ] v7.0 (TiDB 7.0 versions)
- [ ] v6.6 (TiDB 6.6 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:

https://github.com/pingcap/docs/pull/13583

- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: tidb-configuration-file.md [hunk 1]
```diff
@@ -287,6 +287,13 @@ TiDB 配置文件比命令行参数支持更多的选项。你可以在 [config/
 + 默认值：10000
 + 当查询的行数（包括中间结果，基于统计信息）大于这个值，该操作会被认为是 `expensive` 查询，并输出一个前缀带有 `[EXPENSIVE_QUERY]` 的日志。
 
+### `timeout` <span class="version-mark">从 v7.1.0 版本开始引入</span>
+
++ 如果 tidb 写日志卡住，超过这里配置的超时时间，则让 tidb 进程崩溃
++ 默认值: 0 (0 表示不设置超时)
++ 单位：秒
++ 在一些用户场景，tidb 的日志所在的磁盘是热插拔的盘或者通过网络挂载的盘，磁盘设备可能会永久丢失，这种场景下 tidb 无法自动恢复，会在写日志操作上永久阻塞。tidb 进程看起来还活着，但是又不响应请求。这项配置用于处理这样的场景。
```
Comment (Oreoxmt) [inline tidb-configuration-file.md:295]:
```suggestion
+ 用于设置 TiDB 写日志操作的超时时间。当磁盘故障导致日志无法写入时，该配置可以让 TiDB 进程崩溃而不是卡死。
+ 默认值：0，表示不设置超时
+ 单位：秒
+ 在某些用户场景中，TiDB 日志可能是保存在热插拔盘或网络挂载盘上，这些磁盘可能会永久丢失。在这种场景下，TiDB 无法自动恢复，写日志操作会永久阻塞。尽管 TiDB 进程看起来仍在运行，但不会响应任何请求。该配置项用于处理这样的场景。
```

**Review by ran-huang — APPROVED** [review]

**Review by Oreoxmt — APPROVED** [review]

**Comment by Oreoxmt** [comment]
@bb7133 PTAL

**Review by bb7133 — APPROVED** [review]
LGTM

**Pull request merged**